### PR TITLE
Fix some mistakes + clean and add basic tests

### PR DESCRIPTION
**Pull request by sahid**

### CODE REVIEW

File: os_vif/exception.py [hunk 1]
```diff
@@ -22,10 +22,17 @@ class ExceptionBase(Exception):
 
     """
     msg_fmt = _("An unknown exception occurred.")
+    code = 500
```
Comment (jaypipes) [inline os_vif/exception.py]: These are not HTTP exceptions. There is no reason to add this from Nova.

File: os_vif/__init__.py [hunk 2]
```diff
@@ -26,10 +26,9 @@
 
 
 def initialize(reset=False, **config):
-    """
-    Loads all os_vif plugins and initializes them with a dictionary of
-    configuration options. These configuration options are passed as-is
-    to the individual VIF plugins that are loaded via stevedore.
+    """Loads all os_vif plugins and initializes them with a dictionary of
```
Comment (jaypipes) [inline os_vif/__init__.py]: Would it be possible to just disable checking for this (extremely silly IMHO) flake8 rule?

File: os_vif/tests/base.py [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+            try:
+                x.stop()
+            except AssertionError:
+                pass
+
+        # Delete attributes that don't start with _ so they don't pin
+        # memory around unnecessarily for the duration of the test
+        # suite
+        for key in [k for k in self.__dict__.keys() if k[0] != '_']:
+            del self.__dict__[key]
```
Comment (jaypipes) [inline os_vif/tests/base.py]: Please remove the above method. It's not useful for os_vif testng.

File: os_vif/tests/base.py [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+
+        """
+        if not new_attr:
+            new_attr = mock.Mock()
+        patcher = mock.patch.object(obj, attr_name, new_attr, **kwargs)
+        patcher.start()
+        self.addCleanup(patcher.stop)
+
+    # Useful assertions
+    def assertDictMatch(self, d1, d2, approx_equal=False, tolerance=0.001):
```
Comment (jaypipes) [inline os_vif/tests/base.py]: This isn't used.

File: os_vif/tests/base.py [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+        # suite
+        for key in [k for k in self.__dict__.keys() if k[0] != '_']:
+            del self.__dict__[key]
+
+    def log_level(self, level):
+        """Set logging level to the specified value."""
+        log_root = logging.getLogger(None).logger
+        log_root.setLevel(level)
+
+    def mock_object(self, obj, attr_name, new_attr=None, **kwargs):
```
Comment (jaypipes) [inline os_vif/tests/base.py]: this isn't used.

File: os_vif/tests/base.py [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+            except AssertionError:
+                pass
+
+        # Delete attributes that don't start with _ so they don't pin
+        # memory around unnecessarily for the duration of the test
+        # suite
+        for key in [k for k in self.__dict__.keys() if k[0] != '_']:
+            del self.__dict__[key]
+
+    def log_level(self, level):
```
Comment (jaypipes) [inline os_vif/tests/base.py]: this isn't used.

File: os_vif/tests/base.py [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+
+        test_timeout = os.environ.get('OS_TEST_TIMEOUT', 0)
+        try:
+            test_timeout = int(test_timeout)
+        except ValueError:
+            # If timeout value is invalid do not set a timeout.
+            test_timeout = 0
+        if test_timeout > 0:
+            self.useFixture(fixtures.Timeout(test_timeout, gentle=True))
+        self.useFixture(fixtures.NestedTempfile())
```
Comment (jaypipes) [inline os_vif/tests/base.py]: not used... please remove.

File: os_vif/tests/base.py [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+        self.useFixture(fixtures.TempHomeDir())
```
Comment (jaypipes) [inline os_vif/tests/base.py]: not used... please remove

File: os_vif/__init__.py (hunk 2 above)
Comment (sahid) [inline os_vif/__init__.py]: It's not so silly, doc generator uses this line as title

https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings

Like a commit message, we use conventions...

Comment (jaypipes) [inline os_vif/__init__.py]: @sahid You should read the above link. :) It specifically says "The summary line may be on the same line as the opening quotes or on the next line."

The forcing of the summary line to be on the same line as the opening quotes is the _silly_ thing.

**Comment by jaypipes** [comment]
Thank you @sahid!

**Pull request merged**